src/main/java/com/exasol/util/QuotesApplier.java

```diff
@@ -0,0 +1,61 @@
+package com.exasol.util;
+
+import com.exasol.sql.rendering.StringRendererConfig;
```

Comment:
This completely breaks the dependency principles.

Reply:
Splitting at points in Strings is not the safe way to go. We need to know which parts an database object key consists of and apply quoting on the parts. This whole concept needs rework.

Reply:
I extracted this class because I need these methods in two places.